src/Features/CSharp/Portable/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionDiagnosticAnalyzer.Analyzer.cs

```diff
@@ -227,14 +227,39 @@ public override SyntaxKind VisitAssignmentExpression(AssignmentExpressionSyntax
             }
 
             public override SyntaxKind VisitExpressionStatement(ExpressionStatementSyntax node)
+                => IsRef(node.Expression) ? default : Visit(node.Expression);
+
+            public override SyntaxKind VisitReturnStatement(ReturnStatementSyntax node)
             {
-                return Visit(node.Expression);
+                // A "return" statement's expression will be placed in the switch arm expression. We
+                // also can't convert a switch statement with ref-returns to a switch-expression
+                // (currently). Until the language supports ref-switch-expressions, we just disable
+                // things.
+                return node.Expression is null || IsRef(node.Expression)
+                    ? default
+                    : SyntaxKind.ReturnStatement;
             }
 
-            public override SyntaxKind VisitReturnStatement(ReturnStatementSyntax node)
+            private bool IsRef(ExpressionSyntax expression)
             {
-                // A "return" statement's expression will be placed in the switch arm expression.
-                return node.Expression is null ? default : SyntaxKind.ReturnStatement;
+                expression = expression.WalkDownParentheses();
+                if (expression is RefExpressionSyntax)
+                {
+                    return true;
+                }
+
+                if (expression is ConditionalExpressionSyntax conditionalExpression)
```

Comment:
I think this is unnecessary because `ref b ? ref i : ref j` is still a ref expression.

Reply:
💡 You could use 'convert to switch' here.

Reply:
1. are you sure it's unnecessary?  The following compiles just fine for me:

```c#
    static void GetRef(int[] mem, int addr, int mode)
    {
        ref int i = ref addr;
        i = true ? ref mem[mem[addr]] : ref mem[mem[addr]];
```

2. Not sure what you mean by: use 'convert to switch' here.  Can you clarify?

Reply:
Yeah it's correct, there are actually situations where this could happen so nevermind it.

I meant you could switch over expression
`switch (expression) ...` instead of ifs.